templates/firm/show.json.twig

```diff
@@ -0,0 +1,51 @@
+{
+    "@context": "https://schema.org",
+    "@graph": [
+        {
+            "@type": "Organization"
+            {% if firm.name %},"name": "{{firm.name}}" {% endif %}
+            {% if firm.startDate %}, "foundingDate": "{{ firm.startDate }}" {% endif %}
+            {% if firm.endDate %} ,"dissolutionDate": "{{ firm.endDate }}" {% endif %},
+            "@id": "{{ url('firm_show', {'id':firm.id}) }}",
+            "address": {
+                "@type": "PostalAddress"
+                {% if firm.city %},"addressLocality": "{{ firm.city }}" {% endif %}
+                {% if firm.streetAddress %}, "streetAddress": "{{ firm.streetAddress }}" {% endif %}
+            }
+        }
+        {% for titleFirmRole in firm.getTitleFirmRoles("Publisher") %}
+            {% if titleFirmRole.getFirmrole.getName == "Publisher" %}
+                {% set book = titleFirmRole.title %}
+                {% if book %}
+                ,{
+                    "@type": "Book",
+                    "@id": "{{ url('title_show', {'id':book.id}) }}",
+                    "name": "{{book.title}}"
+                    {% if book.pubdate %}, "datePublished": "{{ book.pubdate }}" {% endif %}
+                    {% if book.pubdate %}, "locationCreated": "{{ book.locationOfPrinting }}" {% endif %},
+                    "author": [
+                        {% for titleRole in book.getTitleRoles("Author") %}{% set author = titleRole.person %}
+                        {
+                            "@type": "Person"
+                            {% if (author.firstName) and (author.lastName) %},"name": "{{ author.firstName }} {{ author.lastName }}"  {% endif %},
```

Comment:
There are people with a first name and no last name, and also people with no first name but a last name. So that's two additional cases to include.

Or you can say `{{ author }}` and the __toString() method on person will be called automatically and handle all the cases.